IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -260,38 +260,23 @@ - (MSIDIdToken *)getIDTokenForAccount:(MSIDAccountIdentifier *)account
 {
     MSID_LOG_VERBOSE(context, @"(Default accessor) Get accounts with environment %@, clientId %@, familyId %@", environment, clientId, familyId);
 
-    NSMutableSet *filteredAccountsSet = [NSMutableSet set];
-
     MSIDTelemetryCacheEvent *event = [MSIDTelemetry startCacheEventWithName:MSID_TELEMETRY_EVENT_TOKEN_CACHE_LOOKUP context:context];
 
     NSArray<NSString *> *environmentAliases = [_factory defaultCacheAliasesForEnvironment:environment];
-    __auto_type accountsPerUserId = [self getAccountsPerUserIdForAliases:environmentAliases context:context error:error];
 
-    if (![accountsPerUserId count])
+    NSMutableSet *filteredAccountsSet = [self getAccountsForEnvironment:environment
+                                                     environmentAliases:environmentAliases
+                                                                context:context
+                                                                  error:error];
+
+    if (!filteredAccountsSet)
     {
         MSID_LOG_INFO(context, @"No accounts found, returning!");
         [MSIDTelemetry stopCacheEvent:event withItem:nil success:NO context:context];
         return @[];
     }
 
-    MSIDDefaultCredentialCacheQuery *credentialsQuery = [MSIDDefaultCredentialCacheQuery new];
-    credentialsQuery.credentialType = MSIDRefreshTokenType;
-    credentialsQuery.clientId = clientId;
-    credentialsQuery.familyId = familyId;
-    credentialsQuery.clientIdMatchingOptions = MSIDSuperSet;
-    credentialsQuery.environmentAliases = environmentAliases;
-
-    NSArray<MSIDCredentialCacheItem *> *resultCredentials = [_accountCredentialCache getCredentialsWithQuery:credentialsQuery legacyUserId:nil context:context error:error];
-
-    for (MSIDCredentialCacheItem *credentialCacheItem in resultCredentials)
-    {
-        NSArray *accounts = accountsPerUserId[credentialCacheItem.homeAccountId];
-        if (!accounts) continue;
-
-        [filteredAccountsSet addObjectsFromArray:accounts];
-    }
-
-    if ([resultCredentials count] == 0)
+    if ([filteredAccountsSet count] == 0)
```

Comment:
shouldn't the nil check and count=0 check be reversed here?
count = 0 to indicate no accounts found, and nil to indicate failure?

Reply:
For telemetry purposes we treat both no accounts found and failed lookups as a failure.
The difference is that for failed lookup we don't try to lookup wipe telemetry, because there's something very wrong with our keychain state. 
If we don't find any accounts, but lookup was successful, we also try to read the wipe telemetry to see who removed our accounts.

Reply:
I have the same impression at first glance. 
Probably we could add something to the log info at line 274, like "Fail to read accounts, returning!"? :)

Reply:
Yeah, I think the log line was confusing. I reworded it now :)